src/python/pants/engine/internals/build_files.py

```diff
@@ -240,15 +274,22 @@ async def find_build_file(request: BuildFileAddressRequest) -> BuildFileAddress:
 
 
 @rule
-async def find_target_adaptor(request: TargetAdaptorRequest) -> TargetAdaptor:
+async def find_target_adaptor(
+    request: TargetAdaptorRequest,
+    maybe_build_file_rules_implementation: MaybeBuildFileDependencyRulesImplementation,
+) -> TargetAdaptor:
     """Hydrate a TargetAdaptor so that it may be converted into the Target API."""
     address = request.address
     if address.is_generated_target:
         raise AssertionError(
             "Generated targets are not defined in BUILD files, and so do not have "
             f"TargetAdaptors: {request}"
         )
-    address_family = await Get(AddressFamily, AddressFamilyDir(address.spec_path))
+    address_family, *maybe_origin_family = await MultiGet(
+        Get(AddressFamily, AddressFamilyDir(addr.spec_path))
```

Comment:
Ah! To get the dependency rules in the inference layer, I would need to get the AddressFamily. Except that's under the pants.engine.internals module. Is it ok for plugins to use that?

Once I have the AddressFamily, I can (like below) call `action = build_file_dependency_rules_class.check_dependency_rules(...)` without calling `action.execute()`. Then, if there is at least one module owner that matches an allow rule, then dep inference can discard all other module owners (that match warn or deny rules).

Reply:
Maybe a utility rule could be used to both get stuff here and in anywhere else people want to get the dependency rules actions.

Reply:
Agreed, let's work out how we want to expose the dependency rules for Plugin API consumption.

Reply:
@cognifloyd OK, I've pushed a refactoring to how the dep rule actions are enforced, so you can get the list of actions per dependency without enforcement for your own validations/actions based on that. How's that looking?

Reply:
I think you can mark this as resolved now. Very nice.